### DISCUSSION TITLE
Anyone used GraalVM with the CDK already?

### DISCUSSION THREAD

**Discussion by egonw** [cdk/cdk#1003] [Q&A]
Did anyone try GraalVM for the CDK already? A long time ago, I played with `gcj` to get native binaries for GNU/Linux. The reports on how much faster it is, is encouraging to pick up native compilation again.

I'm asking if anyone did this already, because it would be nice to have a write up of how to use it to process the ChEMBL SDF and see the performance difference.

**Comment by johnmay** [comment]
I think we (and our dependencies) use reflection too much to take full advantage. It works well for small/tight code which can be nicely optimised but most of my previous experiments showed comparable/worse performance.- JohnOn 26 Aug 2023, at 13:31, Egon Willighagen ***@***.***> wrote:﻿
Did anyone try GraalVM for the CDK already? A long time ago, I played with gcj to get native binaries for GNU/Linux. The reports on how much faster it is, is encouraging to pick up native compilation again.
I'm asking if anyone did this already, because it would be nice to have a write up of how to use it to process the ChEMBL SDF and see the performance difference.
